### PR TITLE
Bugfix/web week start ala

### PR DESCRIPTION
**Pull request by oerp-odoo**
Description of the issue/feature this PR addresses:
This PR addresses enterprise ticket `1916666` (in odoo.com)

Current behavior before PR:
Days in calendar view are put into incorrect weeks if Sunday is used as first day of week.

Desired behavior after PR is merged:
Days in calendar view are put into correct weeks regardless which day is first day of week.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/calendar/calendar_model.js [hunk 1]
```diff
@@ -404,7 +405,9 @@ return AbstractModel.extend({
             monthNamesShort: moment.monthsShort(),
             dayNames: moment.weekdays(),
             dayNamesShort: moment.weekdaysShort(),
-            firstDay: _t.database.parameters.week_start,
+            // Index for Sunday is 0 in calendar, but week_start index for
+            // Sunday is 7, so we map it to calendar indexing here.
+            firstDay: week_start != 7 ? week_start : 0,
```
Comment (len-odoo) [inline addons/web/static/src/js/views/calendar/calendar_model.js]: You could have everything on one line:
`firstDay: _t.database.parameters.week_start % 7,  // calendar uses index 0 for Sunday but Odoo stores it as 7`

**Comment by oerp-odoo** [comment]
@len-odoo I amended commit with what you asked (added `opw 1916666` in commit message and made fix as one-liner).

**Comment by oerp-odoo** [comment]
@len-odoo after making changes you suggested, runbot JS tests started to fail. So I had to change it back to original code. Then tests started passing again. Don't know why that specific change triggers tests to fail, cause it looks like it should do same thing as my original code.

Here you can see what was failing in this build (with change you suggested): http://runbot.odoo.com/runbot/build/419944

```
"QUnit test failed: "Views > CalendarView > simple calendar rendering" [message: "Died on test #7     at Object.<anonymous> (http://127.0.0.1:8069/web/static/tests/views/calendar_tests.js:127:11)
    at Object.module (http://127.0.0.1:8069/web/static/lib/qunit/qunit-2.3.0.js:2620:20)
    at http://127.0.0.1:8069/web/static/tests/views/calendar_tests.js:32:7
    at process_job (http://127.0.0.1:8069/web/content/394-21edf6a/web.assets_common.js:3462:274)
    at Object.process_jobs (http://127.0.0.1:8069/web/content/394-21edf6a/web.assets_common.js:3468:49)
    at Object.define (http://127.0.0.1:8069/web/content/394-21edf6a/web.assets_common.js:3451:182)
    at http://127.0.0.1:8069/web/static/tests/views/calendar_tests.js:1:6: Cannot read property 'clone' of undefined"]
```

**Comment by len-odoo** [comment]
The variable might not exist in the test environment...
So we should do 
```
var week_start = _t.database.parameters.week_start || 0;
// calendar uses index 0 for Sunday but Odoo stores it as 7
week_start = week_start % 7;
```
that way we will always give a value to the calendar library instead of `undefined`.
You can check branch 12.0-opw1916666-tst-len, where the runbot is green.

After your fix is merged, I will add a test to make sure the problem does not come back.
This is on branch 12.0-opw1916666-tstx-len (for now the runbot will be red, as your fix isn't merged yet).

Thank you for your investigation.

**Comment by len-odoo** [comment]
robodoo rebase-ff r+

**Comment by len-odoo** [comment]
robodoo retry

**Pull request closed without merging**